packages/sit-onyx/src/components/OnyxButton/OnyxButton.vue

```diff
@@ -0,0 +1,265 @@
+<script lang="ts" setup>
+import { computed } from "vue";
+
+export type ButtonProps = {
+  label?: string;
+  isLoading?: boolean;
+  isDisabled?: boolean;
+  type?: "button" | "submit" | "reset";
+  color?: "primary" | "secondary" | "danger";
+  variant?: "default" | "outline" | "plain";
+  icon?: string;
+};
+
+const props = withDefaults(defineProps<ButtonProps>(), {
+  label: "",
+  isLoading: false,
+  isDisabled: false,
+  type: "button",
+  color: "primary",
+  variant: "default",
+});
+
+const emit = defineEmits<{
+  /**
+   * Emitted when the button is clicked
+   */
+  click: [];
+}>();
+
+const handleClick = () => {
+  emit("click");
+};
+
+const backgroundColor = computed(() => {
+  let resultColor = "transparent";
+
+  if (props.variant === "default") {
+    if (props.color === "primary") {
+      resultColor = "var(--onyx-color-base-primary-500, #0079ce)";
```

Comment:
```suggestion
      resultColor = "var(--onyx-color-base-primary-500)";
```

We don't need all of those fallback values for the CSS variables

Reply:
I've re-written the css.